### PR TITLE
feature/DGJ-609_application-delete

### PR DESCRIPTION
**Pull request by iman-jamali-fw**
## Summary

This PR will allow users to delete an application.
## Changes
- Once a draft is submitted, it will be hard deleted to remove references between the draft and the application.
- Deleting an application would
  - Delete its submission object from Fromio
  - Delete its data in ODS
- On the UI, only applications that have a specific associated processName would have the delete option (in this case, the process name is `senior-leadership-review`)
- Loading indicator was added for application and draft deletion. 
- unrelated to this PR BPM listener for updating and deleting data from ODS by applicationId was added

## Screenshots (if applicable)
Screen Shot 2022-10-21 at 11 19 31 AM

### CODE REVIEW

File: forms-flow-api/src/formsflow_api/resources/application.py [hunk 1]
```diff
@@ -386,3 +388,25 @@ def get():
             )
         except BusinessException as err:
             return err.error, err.status_code
+
+@cors_preflight("DELETE, OPTIONS")
+@API.route("/<int:application_id>/delete", methods=["DELETE", "OPTIONS"])
```
Comment (iman-jamali-fw) [inline forms-flow-api/src/formsflow_api/resources/application.py:393]: I wasn't sure how to name this endpoint.
Delete version of `applications/<applicatoinId>` is already used for Camunda deleting application so I added another one `applications/<applicatoinId>/delete`. Feel free to suggest a better naming :)

File: forms-flow-api/src/formsflow_api/services/application.py [hunk 2]
```diff
@@ -360,15 +362,6 @@ def update_application(application_id: int, data: Dict, **kwargs):
...
             raise BusinessException("Invalid application", HTTPStatus.BAD_REQUEST)
-    
-    @staticmethod
-    def delete_application(application_id: int):
-        """Delete application."""
-        application = Application.find_by_id(application_id=application_id)
-        if application:
-            application.delete()
-        else:
-            raise BusinessException("Invalid application", HTTPStatus.BAD_REQUEST)
```
Comment (iman-jamali-fw) [inline forms-flow-api/src/formsflow_api/services/application.py:371]: Just moved this custom method to the end of the file for easier future upgrades.

File: forms-flow-api/src/formsflow_api/services/draft.py [hunk 3]
```diff
@@ -106,22 +106,6 @@ def update_draft(draft_id: int, data, **kwargs):
...
-        draft = Draft.get_by_id(draft_id, user_id)
-        if draft:
-            draft.delete()
-        else:
-            response, status = {
-                "type": "Bad request error",
-                "message": f"Invalid request data - draft id {draft_id} does not exist",
-            }, HTTPStatus.BAD_REQUEST
-            raise BusinessException(response, status)
-
```
Comment (iman-jamali-fw) [inline forms-flow-api/src/formsflow_api/services/draft.py:124]: Moved this custom method to the end of file.

**Review by bhumin-fw — APPROVED** [review]
Looks good to me, Pretty much clean and readable integration (Y)

**Pull request merged**